### PR TITLE
Fix LDAP sync issues

### PR DESCRIPTION
**Pull request by RouganStriker**
* Remove non-existant emit.debug
* Fix LDAP sync not failing on error
* Improve rendering of LDAP sync error messages

Refs #SDE-3896

### CODE REVIEW

**Comment by funkaoshi** [comment]
Seems fine +1

File: sdetools/modules/sync_ldap/ldap_module.py [hunk 1]
```diff
@@ -967,14 +973,26 @@ def sync_new_users_from_ldap(self, untracked_users, directory_users):
...
+            if isinstance(e.message, dict):
+                error_message = "\n".join([
+                    "{0}: {1}".format(
+                        csv_data[start + int(row_id)]['email'],
+                        " ".join([", ".join(error) for error in errors.values()])
+                    )
+                    for row_id, errors in e.message['errors'].items()
+                ])
+
+            update_failures.append("Error creating all users: {0}".format(error_message))
```
Comment (emintham) [inline sdetools/modules/sync_ldap/ldap_module.py]: "Error creating the following users" maybe? "all users" sounds a bit weird.

File: sdetools/modules/sync_ldap/ldap_module.py [hunk 2]
```diff
@@ -967,14 +973,26 @@ def sync_new_users_from_ldap(self, untracked_users, directory_users):
...
         except APIError as e:
-            update_failures.append(str(e))
-            self.emit.debug(e)
-            self.emit.error("Failed to create all users")
+            # Convert the error from a dict to a nicer string
+            error_message = str(e)
+            if isinstance(e.message, dict):
+                error_message = "\n".join([
+                    "{0}: {1}".format(
+                        csv_data[start + int(row_id)]['email'],
```
Comment (emintham) [inline sdetools/modules/sync_ldap/ldap_module.py]: Is it possible to make the row_id an int key on the dict? then we won't need to do the conversion.

**Comment by emintham** [comment]
Two minor comments. +1 otherwise

File: sdetools/modules/sync_ldap/ldap_module.py (hunk 1 above)
Comment (RouganStriker) [inline sdetools/modules/sync_ldap/ldap_module.py]: Good point, fixed

File: sdetools/modules/sync_ldap/ldap_module.py (hunk 2 above)
Comment (RouganStriker) [inline sdetools/modules/sync_ldap/ldap_module.py]: I believe we are using an integer key when loading items into the dict (we get the row_id from enumerate()). Something must be stringifying it somewhere...

Comment (RouganStriker) [inline sdetools/modules/sync_ldap/ldap_module.py]: I'm going to just leave it as-is for now

**Pull request merged**